### PR TITLE
grafana dashboard and recording rules for volume statistics

### PR DESCRIPTION
**Pull request by ghost**
A Grafana dashboard and required recording rules for gluster-prometheus.

### CODE REVIEW

**Review by shtripat — COMMENTED** [review]
The grafana dashboards added here look more oriented to standalone deployment of gluster server.
How same fits in deployment of glusterfs as container native storage use case?
Do we plan to have different grafana dashboard templates for them?

File: prometheus_rules/recording.rules.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+- name: gluster_volume
```
Comment (shtripat) [inline prometheus_rules/recording.rules.yml]: aggregation rules look fine to me and as I understand all these aggregations are at volume level.

**Comment by labazza** [comment]
Yes i have finished validating the aggregation and alerting. The dashboards also reflect the new rules. I will wait new metrics to be available in master to continue.
@shtripat indeed my use case was for gluster on baremetal. IMO the dashboards represent the gluster architecture itself don't think this would change under a Kubernetes deployment.
However for a better overview and additional dashboard for kubernetes or baremetal related can always be created.

**Comment by shtripat** [comment]
@labazza I tried importing this dashboards in this PR in my grafana setup but it fails with error `Tenplate variables could not be initialized: Cannot read property indexOf of undefined`. Also why the cluster name is hard-coded to `prometheus`.
Am I missing something here for testing the dashboards?

**Comment by anmolsachan** [comment]
@shtripat @aravindavk I was going through this PR; while this PR has some good work, Does it make sense to have this on this project ? As of my understanding Gluster-Prometheus is scoped mainly for metric collection.
@labazza Just out of curiosity, can you please attach screenshots of the dashboards as I was also not able to import the dashboards in my setup even after changing the sources.

**Comment by labazza** [comment]
Since then i have re-factored the dashboards in cosideration @shtripat comment. Please note that some graphs make use of telegraf as well.
Cluster overview:
cluster
Node overview:
node_pt1
node_pt2
Volume overview:
volume

**Comment by cloudbehl** [comment]
@labazza Thanks for PR.

As the grafana dashboard and recording rules for gluster are handled as a part of [gluster-mixins](https://github.com/gluster/gluster-mixins/) project.  It will be good if you can contribute to gluster-mixins project. 

**Gluster Dashboard:**
There are multiple [Issues](https://github.com/gluster/gluster-mixins/issues?q=is%3Aopen+is%3Aissue+label%3AGrafana%2FUI) which are created for the grafana dashboard. @anmolsachan Is leading that effort and the dashboards are written in jsonnet format(As per the k8s instruction and guidlines). The UX design is also copied to each issue.

**Gluster Alerts and Recording rules**
@umangachapagain Is leading the efforts, he has written few alerts and rules already and we are extenting on that. There are set of issues that has a prefix of `alert` are there in the gluster-mixins repo if you want to have a look.

Both dashboard and alerts are written is jsonnet file format and which get converted to [YAML](https://github.com/gluster/gluster-mixins#manually-generate-configs-and-rules) format for alerts and recording rule and JSON for the grafana dashboard. These can be applied to the standalone prometheus and grafana.

if you want ot generate the k8s object of these YAML and JSON there is a [extras](https://github.com/gluster/gluster-mixins/tree/master/extras) folder which helps to covert to k8s object and the final result is copied in manifests folder which can be applied in k8s/openshift cluster. 


@labazza For more context please refer the gluster-mixins documentation. Thanks again for you efforts and feel free to raise PR and query/Issue to the gluster-mixins project.

**Comment by labazza** [comment]
@cloudbehl thanks i will submit them there

**Comment by LukaszKochaniak** [comment]
Hi @labazza,
do you have stored somewhere dashboards present on screenshot in comment: https://github.com/gluster/gluster-prometheus/pull/63#issuecomment-446879065?
I couldn't find them in this MR and didn't see the in MR gluster-mixins project.
Would you be so kind to share them here?
Thanks!

**Comment by ghost** [comment]
Hi @LukaszKochaniak, unfortunately i do not work with gluster since few years and i do not have stored anything related to it. You can grab the dashboard json files from [this commit ](https://github.com/gluster/gluster-prometheus/pull/63/commits/fb31475f5b9350bac72e9556c03a92d855fe32fe).

**Pull request closed without merging**